### PR TITLE
Fix: named query date parameters aren't behaving as expected

### PR DESCRIPTION
**Pull request by andrewbrazzatti**
Fixes issue where named query date parameters:
1. Default values weren't setting correctly when a parameter isn't passed
2. Parameters of type "days" were not being set correctly

### CODE REVIEW

File: typescript/api/services/NamedQueryService.ts [hunk 1]
```diff
@@ -253,26 +253,27 @@ export module Services {
...
-                    //Going forward in time X number of days
-                    nowDateAddOrSubtract = nowDateAddOrSubtract.add(days, 'days');
-                  } else if(days < 0) {
-                    //This "additional" step makes the code self explanatory
-                    days = days * -1;
-                    //Going backwards in time X number of days
-                    nowDateAddOrSubtract = nowDateAddOrSubtract.subtract(days, 'days');
-                  }
-                  value = nowDateAddOrSubtract.toISOString();
-                } else if(queryParam.format == NamedQueryFormatOptions.ISODate) {
```
Comment (cofiem) [inline typescript/api/services/NamedQueryService.ts:271]: Looks like the ISODate branch is missing?

Comment (andrewbrazzatti) [reply typescript/api/services/NamedQueryService.ts:271]: The unit tests will confirm this but line 256 covers the behaviour for ISODate, no further calculations are needed to be made for ISODate, it just needs to set its default value only for the "days" option do we need to translate it to an ISO date string.

File: typescript/api/services/NamedQueryService.ts [hunk 2]
```diff
@@ -255,26 +253,27 @@ export module Services {
...
+              } else if(days < 0) {
+                //This "additional" step makes the code self explanatory
+                days = days * -1;
+                //Going backwards in time X number of days
+                nowDateAddOrSubtract = nowDateAddOrSubtract.subtract(days, 'days');
+              }
+              value = nowDateAddOrSubtract.toISOString();
+            } 
+
+            query[queryParam.queryType] = value;
```
Comment (cofiem) [inline typescript/api/services/NamedQueryService.ts:274]: It looks like values passed as querystring parameters will now be processed the same way as defaults.
That would be good. Was that the intent?

Comment (andrewbrazzatti) [reply typescript/api/services/NamedQueryService.ts:274]: Yes it is

File: test/unit/services/NamedQueryService.test.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+          queryType: "<=",
+          format: "days",
+          whenUndefined: "defaultValue",
+          defaultValue: "-1"
+        },
+        isoDateParam: {
+          type: "date",
+          queryType: "=>",
+          path: "isoDatePath",
+          format: "isoDate",
```
Comment (cofiem) [inline test/unit/services/NamedQueryService.test.js:102]: Is the format name `isoDate` or `ISODate`? The enum uses `ISODate`.

Perhaps the date branch needs to check the format is one of the enums, and fail if it isn't?

**Review by cofiem — CHANGES_REQUESTED** [review]
Check the `ISODate` enum and consider checking the format is one of the enums, then looks good to me.

File: test/unit/services/NamedQueryService.test.js (hunk 3 above)
Comment (andrewbrazzatti) [reply test/unit/services/NamedQueryService.test.js]: Good pickup. I've changed it to ISODate. the "format" option was added later to allow the days feature, for that reason I think it should default to ISODate when not provided. I've added to the tests a check for that.

File: package.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
   "dependencies": {
     "@researchdatabox/oni-ocfl": "^1.4.0",
     "@researchdatabox/raido-openapi-generated-node": "^0.0.3",
-    "@researchdatabox/redbox-core-types": "^1.4.7",
+    "@researchdatabox/redbox-core-types": "file:core",
```
Comment (cofiem) [inline package.json:9]: I'm not sure what 'file:core' means here. Should this be part of this PR?

**Review by cofiem — APPROVED** [review]
Looks good. Merge #2656 before this PR.

**Pull request merged**